### ISSUE TITLE
Circular Dependency w/ 1.8.12

### ISSUE THREAD

**Issue by mbabker** [GeniusesOfSymfony/WebSocketBundle#292]
Line 22 in the below stack trace is where the recursion starts, the rest is left in place just as a frame of reference.

The problem seems to be that the `gos_web_socket.ws.server` service has a dependency to the `gos_web_socket_server.wamp_application` service, which depends on `gos_web_socket.topic.dispatcher`, which depends on `gos_web_socket.wamp.topic_manager`, which loops back to `gos_web_socket_server.wamp_application`.  This isn't caught in container compilation or an issue when running normal web requests, but firing `bin/console gos:websocket:server` does hit it.

```sh
Call Stack:
    0.0021     358776   1. {main}() <PROJECT>/bin/console:0
    0.0350    1938640   2. Symfony\Bundle\FrameworkBundle\Console\Application->run() <PROJECT>/bin/console:26
    0.0612    2253168   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() <PROJECT>/vendor/symfony/console/Application.php:148
    0.1222    5034664   4. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() <PROJECT>/vendor/symfony/framework-bundle/Console/Application.php:72
    0.1222    5034664   5. Symfony\Bundle\FrameworkBundle\Console\Application->find() <PROJECT>/vendor/symfony/console/Application.php:229
    0.2312    6791048   6. Symfony\Bundle\FrameworkBundle\Console\Application->find() <PROJECT>/vendor/symfony/framework-bundle/Console/Application.php:94
    0.2313    6799784   7. Symfony\Bundle\FrameworkBundle\Console\Application->get() <PROJECT>/vendor/symfony/console/Application.php:653
    0.2313    6799784   8. Symfony\Bundle\FrameworkBundle\Console\Application->get() <PROJECT>/vendor/symfony/framework-bundle/Console/Application.php:104
    0.2313    6799784   9. Symfony\Bundle\FrameworkBundle\Console\Application->has() <PROJECT>/vendor/symfony/console/Application.php:483
    0.2313    6799784  10. Symfony\Component\Console\CommandLoader\ContainerCommandLoader->get() <PROJECT>/vendor/symfony/console/Application.php:512
    0.2313    6799784  11. Symfony\Component\DependencyInjection\ServiceLocator->get() <PROJECT>/vendor/symfony/console/CommandLoader/ContainerCommandLoader.php:37
    0.2313    6800160  12. ContainerMqr9iss\appProdProjectContainer->{closure:<PROJECT>/var/cache/prod/ContainerMqr9iss/getConsole_CommandLoaderService.php:104-106}() <PROJECT>/vendor/symfony/dependency-injection/ServiceLocator.php:64
    0.2314    6800480  13. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getConsole_CommandLoaderService.php:105
    0.2316    6804232  14. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_WebsocketServer_CommandService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.2319    6816904  15. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_WebsocketServer_CommandService.php:8
    0.2321    6819024  16. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_EntryPointService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.2323    6825248  17. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_EntryPointService.php:8
    0.2325    6827680  18. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Server_RegistryService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.2327    6830952  19. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Server_RegistryService.php:10
    0.2501    6843280  20. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Ws_ServerService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.2633    7395424  21. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Ws_ServerService.php:8
    0.2636    7406472  22. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocketServer_WampApplicationService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5385   17131912  23. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocketServer_WampApplicationService.php:14
    0.5387   17138056  24. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Topic_DispatcherService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5444   17272056  25. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Topic_DispatcherService.php:8
    0.5447   17273960  26. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Wamp_TopicManagerService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5452   17291120  27. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Wamp_TopicManagerService.php:8
    0.5453   17300984  28. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocketServer_WampApplicationService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5453   17301144  29. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocketServer_WampApplicationService.php:14
    0.5454   17306712  30. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Topic_DispatcherService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5454   17306840  31. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Topic_DispatcherService.php:8
    0.5454   17308424  32. require('<PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Wamp_TopicManagerService.php') <PROJECT>/var/cache/prod/ContainerMqr9iss/appProdProjectContainer.php:1564
    0.5454   17308504  33. ContainerMqr9iss\appProdProjectContainer->load() <PROJECT>/var/cache/prod/ContainerMqr9iss/getGosWebSocket_Wamp_TopicManagerService.php:8
```

**Comment by mbabker** [comment]
FWIW, this works on 1.8.11 and using the forked version of Ratchet because of https://github.com/GeniusesOfSymfony/Ratchet/commit/16c876aecaeb6769aedc3315c9a64cdaa34e6d89

Because of the constructor injection versus a setter, that causes everything to fall apart, more specifically the removal of [this line](https://github.com/GeniusesOfSymfony/WebSocketBundle/pull/283/files#diff-c69c1d1d56b1500c73986fa8d86c0759L123) in the PR moving away from the forked version of Ratchet.

**Comment by mbabker** [comment]
Fixed by #296
